src/views/virtualmachines/list/hooks/useVMListFilters/useVMListFilters.ts

```diff
@@ -1,96 +1,86 @@
+import { getMemoryFilter } from '../../utils/filters/getMemoryFilter';
+import { getNADsFilter } from '../../utils/filters/getNADsFilter';
+import { getOSFilter } from '../../utils/filters/getOSFilter';
+import { getSchedulingFilter } from '../../utils/filters/getSchedulingFilter';
+import { getStatusFilter } from '../../utils/filters/getStatusFilter';
+import { useVirtualMachineInstanceMapper } from '../useVirtualMachineInstanceMapper';
 
-import { useCPUFilter } from './useCPUFilter';
-import { useDateFilter } from './useDateFilter';
-import { useDescriptionFilter } from './useDescriptionFilter';
-import { useInstanceTypesFilter } from './useInstanceTypesFilter';
-import { useIPSearchFilter } from './useIPSearchFilter';
-import { useLiveMigratableFilter } from './useLiveMigratableFilter';
-import { useMemoryFilter } from './useMemoryFilter';
 import { useNodesFilter } from './useNodesFilter';
-import { useOSFilter } from './useOSFilter';
 import { useProjectFilter } from './useProjectFilter';
-import { useStatusFilter } from './useStatusFilter';
 import { useStorageClassFilter } from './useStorageClassFilter';
-import { useTemplatesFilter } from './useTemplatesFilter';
 
 export const useVMListFilters = (
-  vmis: V1VirtualMachineInstance[],
-  vms: V1VirtualMachine[],
   vmims: V1VirtualMachineInstanceMigration[],
   pvcMapper: PVCMapper,
 ): {
-  advancedFilters: RowFilter<V1VirtualMachine>[];
-  filters: RowFilter<V1VirtualMachine>[];
-  projectFilter: RowFilter<V1VirtualMachine>;
-  searchFilters: RowFilter<V1VirtualMachine>[];
+  filtersWithSelect: RowFilter<V1VirtualMachine>[];
+  hiddenFilters: RowFilter<V1VirtualMachine>[];
   vmiMapper: VMIMapper;
   vmimMapper: VMIMMapper;
 } => {
-  const vmiMapper: VMIMapper = useMemo(() => {
-    return (Array.isArray(vmis) ? vmis : [])?.reduce(
-      (acc, vmi) => {
-        const name = vmi?.metadata?.name;
-        const namespace = vmi?.metadata?.namespace;
-        if (!acc.mapper[namespace]) {
-          acc.mapper[namespace] = {};
-        }
-        acc.mapper[namespace][name] = vmi;
-        const nodeName = vmi?.status?.nodeName;
-        if (nodeName && !acc?.nodeNames?.[nodeName]) {
-          acc.nodeNames[nodeName] = {
-            id: nodeName,
-            title: nodeName,
-          };
-        }
-        return acc;
-      },
-      { mapper: {}, nodeNames: {} },
-    );
-  }, [vmis]);
+  const vmiMapper = useVirtualMachineInstanceMapper();
 
   const vmimMapper: VMIMMapper = useMemo(() => getLatestMigrationForEachVM(vmims), [vmims]);
 
-  const statusFilter = useStatusFilter();
-  const templatesFilter = useTemplatesFilter(vms);
-  const osFilters = useOSFilter();
-  const nodesFilter = useNodesFilter(vmiMapper);
-  const liveMigratableFilter = useLiveMigratableFilter();
-  const instanceTypesFilter = useInstanceTypesFilter(vms);
-  const storageClassFilters = useStorageClassFilter(vms, pvcMapper);
+  const [vms] = useK8sWatchResource<V1VirtualMachine[]>({
```

Comment:
You should use `useKubevirtWatchData` hook here to fetch from different clusters